content/en/tracing/trace_collection/custom_instrumentation/android/otel.md

```diff
@@ -0,0 +1,514 @@
+    }
+   ```
+   When writing your application, you can enable development logs by calling the `setVerbosity` method. All internal messages in the library with a priority equal to or higher than the provided level are then logged to Android's Logcat:
+   ```kotlin
+   Datadog.setVerbosity(Log.INFO)
+   ```
+
+3. Configure and enable Trace feature:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+val traceConfig = TraceConfiguration.Builder().build()
+Trace.enable(traceConfig)
+```
+{{% /tab %}}
+
+{{% tab "Java" %}}
+```java
+final TraceConfiguration traceConfig = TraceConfiguration.Builder().build();
+Trace.enable(traceConfig);
+```
+{{% /tab %}}
+{{< /tabs >}}    
+
+4. Datadog tracer implements the [Open Telemetry standard][18]. Create the `OtelTracerProvider` and register the `OpenTelemetrySdk` in `GlobalOpenTelemetry` as a singleton instance to be used in your application. We recommend for best practices to do this in your `onCreate()` method.
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+GlobalOpenTelemetry.set(object : OpenTelemetry {
+    private val tracerProvider = OtelTracerProvider.Builder()
+        .setService([BuildConfig.APPLICATION_ID])
+        .build()
+
+    override fun getTracerProvider(): TracerProvider {
+        return tracerProvider
+    }
+
+    override fun getPropagators(): ContextPropagators {
+        return ContextPropagators.noop()
+    }
+})
+// and later on if you want to access the tracer provider
+val tracerProvider = GlobalOpenTelemetry.get().getTracer(instrumentationName = "<instrumentation_name>")
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+
+{{% /tab %}} 
+{{% tab "Java" %}}
+```java
+GlobalOpenTelemetry.set(new OpenTelemetry() {
+    private final TracerProvider tracerProvider = new OtelTracerProvider.Builder()
+            .setService(BuildConfig.APPLICATION_ID)
+            .build();
+
+    @Override
+    public TracerProvider getTracerProvider() {
+        return tracerProvider;
+    }
+
+    @Override
+    public ContextPropagators getPropagators() {
+        return ContextPropagators.noop();
+    }
+};
+// and later on if you want to access the tracer provider
+final TracerProvider tracerProvider = GlobalOpenTelemetry.get().getTracer("<instrumentation_name>");       
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+**Note**: If you decide to set the `GlobalOpenTelemetry.set` API you need to make sure that this is only called one single time in your process otherwise OpenTelemetry will throw an exception. In case you want to avoid that you can just create a `TracerProvider` and use it as a singleton in your project.
+
+**Note**: The `setService` method is used to set the service name for the tracer provider. The service name is used to identify the application in the Datadog UI. You can either use the `GlobalOpenTelemetry` to hold a single instance of the `TracerProvider` create your own instance and use it in your application code as needed.
```

Comment:
```suggestion
```
I think this note is duplicated?

Reply:
will double check, tks !!

Reply:
I did not find any duplicate, can you please double check it on your end ?

Reply:
Hey, apologies for not being clearer. I was referring to the overlap between the note on line 382 and 384. Perhaps these could be combined?

```
Ensure the GlobalOpenTelemetry.set API is called only once per process. The setService method is used to set the service name for the tracer provider, which identifies the application in Datadog.

You have two options for managing the TracerProvider instance:

- Use GlobalOpenTelemetry to hold a single instance of the TracerProvider. This ensures that the GlobalOpenTelemetry.set API is called only once per process.
- Create your own instance of the TracerProvider and use it as a singleton in your project. This allows you to manage the instance directly in your application code as needed.
```